### ISSUE TITLE
yolov4 nan average loss after few iterations.

### ISSUE THREAD

**Issue by kai5gabriel** [AlexeyAB/darknet#7257]
I am using the yolov4 for training on my custom dataset. Initially I had used the same repo for tiny yolov3 version and it worked fine. But when I tried to train with yolov4 after few iterations i am getting nan as the average loss.
Based on few issues reported earlier I tried to use the above commit 8c9c517 which resolved the issue for others. But I am still seeing the same issue as the original head version.
Originally I had downloaded the code on January 8th 2021. @AlexeyAB  is this issue already resolved, can you please help.

I am using the following command for training:
./darknet detector train cfg/obj.data cfg/yolov4-custom.cfg yolov4.conv.137 -gpus 0,1,2,3 -dont_show -mjpeg_port 8090 -map

I tried without loading the pretrained weights yolov4.conv.137 as well, still the same results.

I am attaching the cfg file for reference.
[yolov4-custom.txt](https://github.com/AlexeyAB/darknet/files/5829435/yolov4-custom.txt)

Error:
v3 (iou loss, Normalizer: (iou: 0.07, cls: 1.00) Region 161 Avg (IOU: 0.000000, GIOU: 0.000000), Class: nan, Obj: 0.000000, No Obj: 0.000000, .5R: 0.000000, .75R: 0.000000, count: 23, class_loss = 5.500000, iou_loss = 0.000000, total_loss = 5.500000
 total_bbox = 6213065, rewritten_bbox = 0.584961 %
v3 (iou loss, Normalizer: (iou: 0.07, cls: 1.00) Region 139 Avg (IOU: 0.000000, GIOU: -0.076910), Class: 0.000000, Obj: 0.000000, No Obj: 0.000000, .5R: 0.000000, .75R: 0.000000, count: 109, class_loss = 53.499996, iou_loss = 0.000000, total_loss = 53.499996
v3 (iou loss, Normalizer: (iou: 0.07, cls: 1.00) Region 139 Avg (IOU: 0.000000, GIOU: -0.007035), Class: 0.000000, Obj: 0.000000, No Obj: 0.000000, .5R: 0.000000, .75R: 0.000000, count: 114, class_loss = 56.999996, iou_loss = 0.000000, total_loss = 56.999996
v3 (iou loss, Normalizer: (iou: 0.07, cls: 1.00) Region 150 Avg (IOU: 0.000000, GIOU: 0.000000), Class: nan, Obj: 0.000000, No Obj: 0.000000, .5R: 0.000000, .75R: 0.000000, count: 27, class_loss = 6.750000, iou_loss = 0.000000, total_loss = 6.750000
v3 (iou loss, Normalizer: (iou: 0.07, cls: 1.00) Region 150 Avg (IOU: 0.000000, GIOU: 0.000000), Class: nan, Obj: 0.000000, No Obj: 0.000000, .5R: 0.000000, .75R: 0.000000, count: 89, class_loss = 22.250000, iou_loss = 0.000000, total_loss = 22.250000
v3 (iou loss, Normalizer: (iou: 0.07, cls: 1.00) Region 161 Avg (IOU: 0.000000, GIOU: 0.000000), Class: nan, Obj: 0.000000, No Obj: 0.000000, .5R: 0.000000, .75R: 0.000000, count: 6, class_loss = 1.500000, iou_loss = 0.000000, total_loss = 1.500000
v3 (iou loss, Normalizer: (iou: 0.07, cls: 1.00) Region 161 Avg (IOU: 0.000000, GIOU: 0.000000), Class: nan, Obj: 0.000000, No Obj: 0.000000, .5R: 0.000000, .75R: 0.000000, count: 2, class_loss = 0.500000, iou_loss = 0.000000, total_loss = 0.500000
 total_bbox = 6229632, rewritten_bbox = 0.583325 %
 total_bbox = 6202249, rewritten_bbox = 0.572776 %
Syncing... Done!

 (next mAP calculation at 6637 iterations)
 Last accuracy mAP@0.5 = 0.00 %, best = 0.00 %
 6260: 40.010807, -nan avg loss, 0.004000 rate, 8.679132 seconds, 1602560 images, 210.229850 hours left
known client: 77, sent = 182923, must be sent outlen = 182923

**Comment by AlexeyAB** [comment]
* This bug with new_coords=1 was only during period [Dec 7, 2020 - Dec 12, 2020]   https://github.com/AlexeyAB/darknet/issues/7109

* Did you train the first 1000 iterations by using 1 GPU? https://github.com/AlexeyAB/darknet#how-to-train-with-multi-gpu


* show screenshots with such information
```
./darknet detector test cfg/coco.data cfg/yolov4.cfg yolov4.weights data/dog.jpg
 CUDA-version: 10000 (10000), cuDNN: 7.4.2, CUDNN_HALF=1, GPU count: 1
 CUDNN_HALF=1
 OpenCV version: 4.2.0
 0 : compute_capability = 750, cudnn_half = 1, GPU: GeForce RTX 2070
net.optimized_memory = 0
mini_batch = 1, batch = 8, time_steps = 1, train = 0
   layer   filters  size/strd(dil)      input                output
   0 conv     32       3 x 3/ 1    608 x 608 x   3 ->  608 x 608 x  32 0.639 BF
```

**Comment by joemanjiyil** [comment]
> 
> 
>     * This bug with new_coords=1 was only during period [Dec 7, 2020 - Dec 12, 2020]   #7109
> 
>     * Did you train the first 1000 iterations by using 1 GPU? https://github.com/AlexeyAB/darknet#how-to-train-with-multi-gpu
> 
>     * show screenshots with such information
> 
> 
> ```
> ./darknet detector test cfg/coco.data cfg/yolov4.cfg yolov4.weights data/dog.jpg
>  CUDA-version: 10000 (10000), cuDNN: 7.4.2, CUDNN_HALF=1, GPU count: 1
>  CUDNN_HALF=1
>  OpenCV version: 4.2.0
>  0 : compute_capability = 750, cudnn_half = 1, GPU: GeForce RTX 2070
> net.optimized_memory = 0
> mini_batch = 1, batch = 8, time_steps = 1, train = 0
>    layer   filters  size/strd(dil)      input                output
>    0 conv     32       3 x 3/ 1    608 x 608 x   3 ->  608 x 608 x  32 0.639 BF
> ```

Okay, I have taken the latest codebase now. I started the training with multi gpus earlier. Now I have trained for a little over 1000 iterations on single gpu and now kept it for training on multi gpus. Training looks to be progressing well now.

Can you explain why a single GPU training is required for initial 1000 iterations.
